addons/account/models/account_invoiceable_line.py

```diff
@@ -0,0 +1,117 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _, _lt
+from odoo.tools import format_date
+
+
+class InvoiceableLineMixin(models.AbstractModel):
+    _name = 'account.order.line'
+    _description = 'Enables downpayment support for child records of account.order that can be down payments'
+
+    is_downpayment = fields.Boolean(string="Is a down payment",
+                                    help="Down payments are made when creating account moves from an order."
+                                         " They are not copied when duplicating an order.")
+
+    # To override
+    invoice_lines = fields.One2many('account.move.line')
+    display_type = fields.Selection([
+        ('line_section', "Section"),
+        ('line_note', "Note")], default=False, help="Technical field for UX purpose.")
```

Comment:
I had a look :monocle_face: 

For POL, it doesn't work because of this line https://github.com/odoo/odoo/blob/17.0/addons/purchase/models/purchase_order_line.py#L197 . If you remove the explicit default in the field declaration, the dict returned by default_get will not contain 'display_type', and you'll get an error. Though, this line seems an absolute nonsense, since the default_get will always return False, so essentially, we'll only enter the condition if values.get('display_type') is evaluated to True. 

For SOL, I noticed your branch was declaring the field a second time with the same values as in account.order.line.mixin ; I guess it's a leftover, you should remove it. But apart from that, I don't have any error when creating them :thinking: